### ISSUE TITLE
Replace Resources drop down choices with Resource group choices in Work Unit filtering section

### ISSUE THREAD

**Issue by muziqaz** [FoldingAtHome/fah-web-client-bastet#195]
![fah resources](https://github.com/user-attachments/assets/b88fe927-7fb0-448c-ada7-51bb4ea10b97)

So at the moment we are given an option to filter by number of CPUs and GPU names. The problem with that is with bigger farms we will have different generation/architecture CPUs with same amount of CPU cores/threads, and when filtering with let's say CPU 15 we will get all the CPU:15 entries, even though I would want results only from one generation/PC/resource group to show up and not all of my CPU:15 machines.

Let's say I have AMD 7950x with CPU:30
and I will have AMD 9950x with also CPU:30
plus one might have an Intel chip with CPU:30

So I want to get a PPD average for AMD 7950x chip, but I wouldn't be able to do so, because all CPU:30 entries would show up under that filter option.
I would name 7950x resource group as 7950x, and other chip groups with their own names, that I can differenciate easily what is folding and how.
So I think it would benefit to have Resource Group column in WU History section, and Resource Group as filtering option.

**Comment by jcoffland** [comment]
Summary: TPF filtered by CPU counts doesn't give a good average with different CPU models.

**Comment by muziqaz** [comment]
Indeed, since CPU:30 filter might show TPF and PPD for 2 different systems with CPU:30 setting

**Comment by muziqaz** [comment]
To add to this, if I remove a GPU from the system, and put another GPU in the same system. V8 WU History does not show that old GPU in the Resources drop down menu. See the screenshot below.
PC3 used to have Radeon VII and RX550. Now it has Intel GPU.
1: Radeon VII and RX550 no longer are available to view
2: If I select Intel GPU from Resources drop down menu, WU History shows all the WU finished by Radeon VII.
Ideally we need to retain historic GPUs in drop-down menu, maybe until WUs which were finished by old GPUs have been removed from WU History due to 500 entry limit in the database

![Screenshot_2024-12-19-14-07-03-561_org mozilla firefox~2](https://github.com/user-attachments/assets/25d88c41-60b4-4e34-87a8-984e9be1a4f1)
